### ISSUE TITLE
[SDK 50] [expo-dev-client] Tunnel has stopped working on iOS

### ISSUE THREAD

**Issue by rayoverweij** [expo/expo#26564]
### Summary

Since upgrading to SDK 50, using `--tunnel` with development builds on iOS has stopped working with the error message "Failed to load app from [tunnel URL] with error: The resource could not be loaded because the App Transport Security policy requires the use of a secure connection."

This happens in both our organisation's repository as well as in a blank project.

Searching the internet for this error gives a couple references to loosening security rules in either Xcode or Info.plist, neither of which I can do from the managed workflow, and nothing related to Expo specifically. It seems that iOS takes issue with loading the data form a http url instead of a https one though as far as I'm aware that has always been the case for tunnel URLs.

I haven't tested with Android yet as I won't have an Android phone at my disposal until I'm back in the office next week but the error seems to be iOS-specific anyway.

(Using development builds with a LAN connection has for whatever reason never worked for me as they always time out - yes I'm on the same network and yes I've tried turning off Windows Firewall, maybe it's a WSL thing, but anyway, that's never been an issue as tunnelling has always worked fine. But with this bug popping up I now have no way of using the dev build anymore so I hope this can be resolved soon.)

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

iOS

### Package versions

```
"dependencies": {
    "expo": "~50.0.2",
    "expo-dev-client": "~3.3.6",
    "expo-status-bar": "~1.11.1",
    "react": "18.2.0",
    "react-native": "0.73.2"
},
"devDependencies": {
    "@babel/core": "^7.20.0"
},
```

### Environment

```
System:
      OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
      Shell: 5.1.16 - /bin/bash
Binaries:
      Node: 18.18.0 - ~/.nvm/versions/node/v18.18.0/bin/node
      npm: 10.1.0 - ~/.nvm/versions/node/v18.18.0/bin/npm
npmPackages:
      expo: ~50.0.2 => 50.0.2
      react: 18.2.0 => 18.2.0
      react-native: 0.73.2 => 0.73.2
Expo Workflow: managed
```

### Reproducible demo

The minimum required steps to set up a blank Expo project with an iOS dev build, that is,
1. `npx create-expo-app`
2. `npx expo install expo-dev-client`
3. `eas login`
4. `eas build --profile development --platform ios`
5. Install the build on local device
6. `APP_VARIANT=development npx expo start --tunnel`
7. Scan the QR code

To satisfy the bot: [https://github.com/expo/expo/tree/main/templates/expo-template-bare-minimum](https://github.com/expo/expo/tree/main/templates/expo-template-bare-minimum) (look, as said, it happens in both my years old repo and a brand-new one, so it's not about the code - either it's an Expo regression or an issue with my environment and a demo repo is not going to clear that up)

**Comment by sampsoninc916** [comment]
I've had this problem as well. Attempted to set the Info.plist to accept http calls, open the 8081 port on my computer and router, and go directly to a development build to solve the problem, but it didn't work. The app does work, however, when I do a complete build for ad hoc distribution for testing, but that kind of build does not allow me to debug the app like I want it to.

**Comment by brentvatne** [comment]
thanks for the report! closing because it's a duplicate of #26564, but we will investigate and follow up there

**Comment by the-simian** [comment]
@brentvatne I believe this might have been closed in error, did you mean to close issue #26563 ? The expo-bot auto-closed that one, so this issue might slip under the radar. I am also experiencing this issue after upgrading to expo version `~50.0.3`. I am experiencing the issue using WSL2 on windows, so the error looks a little different. I get the message: `Tunnel URL not found (it might not be ready yet), falling back to LAN URL.`

This is after installing `@expo/ngrok` both globally as well as locally in the newly scaffolded project as a dev dependency.

I am happy to create a reproduction of this, but it would simple be this 
1. run `create-expo-app reproduce-tunnel-problem --template` and selecting typescript and routing (i don't think this matters, but its what I did)
2. npm i --save-dev @expo/ngrok && npm i -g @expo/ngrok
3. `npx expo start --go --tunnel`

you will get:

```
Starting project at /home/the-simian/reproduce-tunnel-problem
Tunnel URL not found (it might not be ready yet), falling back to LAN URL.
Starting Metro Bundler
Tunnel connected.
Tunnel ready.
```

Shall we reopen this issue or should I submit another issue and reference #26564 and #26563 ?
